timers.c

```diff
@@ -1337,6 +1367,67 @@
     }
 /*-----------------------------------------------------------*/
 
+    #if ( portUSING_GRANULAR_LOCKS == 1 )
+        static void vTimerEnterCritical( void )
+        {
+            portDISABLE_INTERRUPTS();
+            {
+                const BaseType_t xCoreID = ( BaseType_t ) portGET_CORE_ID();
+
+                /* Task spinlock is always taken first */
+                portGET_SPINLOCK( xCoreID, &xTaskSpinlock );
+
+                /* Take the ISR spinlock next */
+                portGET_SPINLOCK( xCoreID, &xISRSpinlock );
+
+                /* Increment the critical nesting count */
+                portINCREMENT_CRITICAL_NESTING_COUNT( xCoreID );
+            }
+        }
+    #endif /* #if ( portUSING_GRANULAR_LOCKS == 1 ) */
+
+/*-----------------------------------------------------------*/
+
+    #if ( portUSING_GRANULAR_LOCKS == 1 )
+        static void vTimerExitCritical( void )
+        {
+            const BaseType_t xCoreID = ( BaseType_t ) portGET_CORE_ID();
+
+            configASSERT( portGET_CRITICAL_NESTING_COUNT( xCoreID ) > 0U );
+
+            /* Get the xYieldPending status inside the critical section. */
+            BaseType_t xYieldCurrentTask = xTaskUnlockCanYield();
```

Comment:
```suggestion
            const BaseType_t xCoreID = ( BaseType_t ) portGET_CORE_ID();

            /* Get the xYieldPending status inside the critical section. */
            BaseType_t xYieldCurrentTask = xTaskUnlockCanYield();

            configASSERT( portGET_CRITICAL_NESTING_COUNT( xCoreID ) > 0U );            
```
Suggest to swap the order to prevent compiler error.
>/home/ubuntu/kernel/granular_lock/coverage/FreeRTOS/FreeRTOS/Source/timers.c: In function ‘vTimerExitCritical’:
/home/ubuntu/kernel/granular_lock/coverage/FreeRTOS/FreeRTOS/Source/timers.c:1399:13: error: ISO C90 forbids mixed declarations and code [-Werror=declaration-after-statement]
 1399 |             BaseType_t xYieldCurrentTask = xTaskUnlockCanYield();
      |             ^~~~~~~~~~